tests/test_optics.py

```diff
@@ -570,13 +543,20 @@ def test_OpticalPSF_pupil_plane():
     ref_psf = galsim.OpticalPSF(lam_over_diam, obscuration=obscuration, nstruts=nstruts,
                                 oversampling=pp_oversampling, strut_angle=strut_angle,
                                 pad_factor=pp_pad_factor)
-    im = galsim.fits.read(os.path.join(imgdir, pp_file))
+    if os.path.isfile(os.path.join(imgdir, pp_file)):
+        im = galsim.fits.read(os.path.join(imgdir, pp_file))
+    else:
+        im = galsim.Image(ref_psf._psf.aper.illuminated.astype(float))
+        im.write(os.path.join(imgdir, pp_file))
+    ref_im_scale = ref_psf._psf.aper.pupil_plane_scale
     test_psf = galsim.OpticalPSF(lam_over_diam, obscuration=obscuration,
                                  oversampling=pp_oversampling, pupil_plane_im=im,
-                                 pad_factor=pp_pad_factor)
+                                 pad_factor=pp_pad_factor,
+                                 _pupil_plane_scale=ref_im_scale)
```

Comment:
Is this `_pupil_plane_scale` something that users will want to set?  

If so, we should probably remove the initial underscore.  And if not, you should add a comment here explaining why it needs to be set for this test (and why the default is what users will want, but for whatever reason doesn't apply to this test).